examples/cryptodata-lib/main.go

```diff
@@ -0,0 +1,91 @@
+//  Copyright (c) 2018 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"github.com/ligato/cn-infra/db/cryptodata"
+	"os"
+	"io/ioutil"
+	"encoding/pem"
+	"crypto/x509"
+	"crypto/rsa"
+)
+
+func main() {
+	// Read private key
+	bytes, err := ioutil.ReadFile("key.pem")
+	if err != nil {
+		panic(err)
+	}
+	block, _ := pem.Decode(bytes)
+	if block == nil {
+		panic("failed to decode PEM for key key.pem")
+	}
+	privateKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err != nil {
+		panic(err)
+	}
+
+	// Read public key
+	bytes, err = ioutil.ReadFile("key-pub.pem")
+	if err != nil {
+		panic(err)
+	}
+	block, _ = pem.Decode(bytes)
+	if block == nil {
+		panic("failed to decode PEM for key key-pub.pem")
+	}
+	pubInterface, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		panic(err)
+	}
+	publicKey := pubInterface.(*rsa.PublicKey)
+
+	// Create cryptodata client
+	client := cryptodata.NewClient(cryptodata.ClientConfig{
+		PrivateKeys: []*rsa.PrivateKey{privateKey},
+	})
+
+	// Pass 1st argument from CLI as string to encrypt
+	input := []byte(os.Args[1])
+	fmt.Printf("Input %v\n", string(input))
+
+	// Encrypt input string using public key
+	encrypted, err := client.EncryptData(input, publicKey)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("Encrypted %v\n", string(encrypted))
+
+	// Decrypt previously encrypted input string
+	decrypted, err := client.DecryptData(encrypted)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("Decrypted %v\n", string(decrypted))
+
+	// Try to decrypt JSON with encrypted data
+	encryptedJSON := fmt.Sprintf(`{ "encrypted": "true", "payload": "$crypto$%v" }`, string(encrypted))
```

Comment:
This should also show an example with encrypted data being deeper than just root level.